tests/spec/config_spec.lua

```
@@ -0,0 +1,301 @@
+    })
+    
+    config.reset()
+    local current_config = config.get()
+    
+    assert.equals(current_config.llm.provider, "openai", "Should reset to default provider")
+    assert.equals(current_config.llm.temperature, 0.7, "Should reset to default temperature")
+  end)
+  
+  it("should validate provider availability", function()
+    config.setup()
+    
```

Luacheck (1.2.0)

[warning] line contains only whitespace

(W611)